File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/工厂模式/java/PC.java
package com.redkey.kotlintest.kotlin.design_pattern.工厂模式.java;

public class PC extends Computer {
    private String ram;
    private String hdd;
    private String cpu;

    protected PC(String ram, String hdd, String cpu) {
        this.ram = ram;
        this.hdd = hdd;
        this.cpu = cpu;
    }

    @Override
    public String getRAM() {
        return this.ram;
    }

    @Override
    public String getHDD() {
        return this.hdd;
    }

    @Override
    public String getCPU() {
        return this.cpu;
    }
}

File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/工厂模式/kotlin/PC.kt
package com.redkey.kotlintest.kotlin.design_pattern.工厂模式.kotlin

class PC(var ram: String, var hdd: String, var cpu: String) : Computer() {
    override fun getRAM(): String {
        return this.ram
    }

    override fun getHDD(): String {
        return this.hdd
    }

    override fun getCPU(): String {
        return this.cpu
    }
}